### DISCUSSION TITLE
How to enable tab completion for alias approach?

### DISCUSSION THREAD

**Discussion by shanehh** [nat-n/poethepoet#67] [Q&A] [answered]
I defined `alias poe='poetry run poe'`, and followed "Enable tab completion for your shell" section, but not works.

Any help would be appreciated.

**Comment by nat-n** [comment, marked as answer]
Hi @shanehh,

Could you describe how is is not working? Which shell type are you trying to enable completions for?

The issue could be related to the fact that the completion script installed in your shell needs to call `poe _list_tasks` to get the present tasks, and might not be able to find the poe executable with the alias!

Since the completion requires poe to be accessible, the most reliable solution is to ensure it's installed into your user environment via pip or pipx. Though if you want to try another solution you can always modify the completion script the poe outputs, to change where it looks for the poe executable when calling `poe _list_tasks`.

**Reply by shanehh** [reply]
Thank you so much! I figure it out. I'm using `oh-my-zsh`, here is my recipe

1. In `~/.zshrc`

```bash
alias poe="poetry run poe"
setopt complete_aliases
```

2. change a line in `~/.oh-my-zsh/completions/_poe`

```bash
local TASKS=($(poetry run poe _list_tasks))
```

reference: https://stackoverflow.com/questions/63534197/tab-completion-for-alias-with-the-same-names-in-zsh
